### PR TITLE
Update for OpenTracing Java v0.31.0-RC1

### PR DESCRIPTION
**Pull request by objectiser**
Signed-off-by: Gary Brown <gary@brownuk.com>

This PR uses OT Java 0.31.0-RC1.

### CODE REVIEW

**Comment by objectiser** [comment]
@pavolloffay @yurishkuro

File: build.gradle [hunk 1]
```diff
@@ -11,9 +11,9 @@ plugins {
     id 'io.codearte.nexus-staging' version '0.11.0'
 }
 
-ext.developmentVersion = '0.21.0-SNAPSHOT'
+ext.developmentVersion = '0.30.0-RC1-SNAPSHOT'
```
Comment (yurishkuro) [inline build.gradle]: 0.22.0-RC1 ?

File: jaeger-core/src/main/java/com/uber/jaeger/Span.java [hunk 2]
```diff
@@ -236,30 +236,21 @@ public Span log(long timestampMicroseconds, Map<String, ?> fields) {
...
+    return log(tracer.clock().currentTimeMicros(), event);
   }
 
   @Override
   public Span log(long timestampMicroseconds, String event) {
-    return log(timestampMicroseconds, event, null);
-  }
-
-  @Override
-  public Span log(String message, /* @Nullable */ Object payload) {
```
Comment (yurishkuro) [inline jaeger-core/src/main/java/com/uber/jaeger/Span.java]: I didn't realize this is being removed in OT - we should only be deprecating in OT

Comment (objectiser) [reply jaeger-core/src/main/java/com/uber/jaeger/Span.java]: Should we raise this, and get a RC2 produced before doing a release?

File: build.gradle (hunk 1 above)
Comment (objectiser) [reply build.gradle]: yep will change.

File: jaeger-core/src/main/java/com/uber/jaeger/Span.java (hunk 2 above)
Comment (yurishkuro) [reply jaeger-core/src/main/java/com/uber/jaeger/Span.java]: Yes, I think we should - it's a bad practice to just remove a method

Comment (pavolloffay) [reply jaeger-core/src/main/java/com/uber/jaeger/Span.java]: https://github.com/opentracing/opentracing-java/issues/189#issuecomment-333241372

**Comment by objectiser** [comment]
@yurishkuro Would it be possible for someone with commit/release privileges to merge this in a separate branch and release it?

**Comment by pavolloffay** [comment]
If we want to release `#.#.#-RC#` then release configuration will need changes.

**Comment by objectiser** [comment]
@yurishkuro @black-adder @vprithvi Should be ready to merge into separate branch and release as tag v0.22.0-RC1

**Comment by pavolloffay** [comment]
@jpkrohling could you please also have a look if the release process will work? 

We want to release `0.22.0-RC1` by pushing a tag `git tag v0.22.0-RC1` in `v0.22.0` branch.

**Comment by objectiser** [comment]
@pavolloffay Juca has already reviewed and said the only change required was the regex pattern.

**Comment by jpkrohling** [comment]
I reserve the right to be wrong, though :)

**Pull request merged**